### ISSUE TITLE
Allow to have templates for index, stories and test files?

### ISSUE THREAD

**Issue by nealoke** [CVarisco/create-component-app#26]
Is it possible to also write templates for `index.js`, `template.stories.js` and `template.spec.js`? When I try it with my current templates it does not output it correctly or am I doing it wrong?

I can't find that much docs on how to do this, nor how to use the `COMPONENT_NAME` in an `import COMPONENT_NAME from './COMPONENT_NAME.js`

**Comment by CVarisco** [comment]
Hi @nealoke !

Thanks a lot for your feedback 🙂 
Could you explain much better what is the problem and what are you doing? 🌴

**Comment by nealoke** [comment]
Waaw that was fast 😄 

Using `COMPONENT_NAME.spec.js` is not working, it simply does not use or copy the file from the template folder.

**`COMPONENT_NAME.spec.js` contents**
```
import React from "react";
import { shallow } from "enzyme";

import COMPONENT_NAME from "./COMPONENT_NAME";

describe("COMPONENT_NAME", () => {
	let component;
	let props;

	beforeEach(() => {
		props = {};
		component = shallow(<COMPONENT_NAME {...props} />);
	});

	it("should", () => {
		expect(component).toMatchSnapshot();
	});
});
```

**Comment by nealoke** [comment]
I'm trying to create my own templates for each file (eg stories / test / index / COMPONENT_NAME.js).

My `config.json` is currently the following:
```
{
  "type": "custom",
  "templatesPath": "ccarc/templates",
  "path": "components/"
}
```

**Comment by CVarisco** [comment]
So, the problem is the naming of the file after the choice of component name?

**Comment by nealoke** [comment]
@CVarisco see updated explanation, it was a mess before sorry!

**Comment by CVarisco** [comment]
@nealoke Thanks a lot for your information!

There is a bug on the copy of the templates files to the destination folder.  
I will investigate in these days to fix as soon as possible the bug.
(If you want to contribute, you are welcome 🌮 )

But for clarification, the `path` value on config file should be `components` without `/` because the destination folder when the script start is `${path}/` [files.js:84](https://github.com/CVarisco/create-component-app/blob/master/src/files.js#L84)

**Comment by nealoke** [comment]
@CVarisco Thank you so very much for this tool and for looking into it :)

I will have a look but I really am not familiar with contributing to projects 😢 . I've stripped the `/` from the path now. 

Also be aware that the `componentName.styles.scss` is also renamed to simply `styles.scss`.

**Comment by nealoke** [comment]
I've figured out where it is going wrong, you mind pushing this to NPM?

The issue lies in the `generateFileName` function. 

**Currently is**
```
function generateFileName(newFilePath, newFileName, fileName) {
  // Suppose that the index file don't be renamed
  if (fileName.indexOf('index') !== -1) {
    return fileName
  }

  if (fs.existsSync(newFilePath)) {
    return fileName
  }

  return newFileName
}
```

So this does not replace the `COMPONENT_NAME` in the `fileName`. A simple if statement solves the issue :)

```
function generateFileName(newFilePath, newFileName, fileName) {
  // Suppose that the index file don't be renamed
  if (fileName.indexOf('index') !== -1) {
    return fileName
  }

  if (fs.existsSync(newFilePath)) {
    return fileName
  }

  if (fileName.includes('COMPONENT_NAME')) {
    return fileName.replace(/COMPONENT_NAME/g, newFileName)
  }

  return newFileName
}


```

**Comment by CVarisco** [comment]
Woooooo 🔝 🎉 

I'm very happy for you 🙂 !
I hope that my project will be the first of a lot of others 🥇 

I published a new version (v0.8.5) on NPM with your fix!
Thanks again and have a nice day 👓
